rfcs/0013-changes-lifecyle.md

```diff
@@ -0,0 +1,348 @@
+- Feature Name: changes-lifecycle
+- Start Date: 2024-08-05
+- RFC PR: [FuelLabs/sway-rfcs#41](https://github.com/FuelLabs/sway-rfcs/pull/41)
+
+# Summary
+
+[summary]: #summary
+
+This RFC is a guide on how to introduce changes following the compiler release lifecycle.
+
+# Motivation
+
+[motivation]: #motivation
+
+To maximize user experience when updating the compiler, we need a guide on how to introduce changes.
+
+# Guide-level explanation
+
+[guide-level-explanation]: #guide-level-explanation
+
+The compiler will follow a "rolling release" scheme, which means that periodically (to be specified) a
+new version will be released.
+
+This means that as soon as the compiler reaches version "1.0.0", the next **version** will be "1.1.0"; and
```

Comment:
following from the [torvolds line](https://lkml.org/lkml/2012/12/23/75), "WE DO NOT BREAK USERSPACE!" Should we clarify that patches will respect semver expectations? ie. minor releases and patches don't require any user intervention? Unless we are pre 1.0, then minor releases are treated like major breaking ones.

Its not clear to me from the rolling release scheme description here what guarantees - _if any_ - we are providing to our users and how the versions signal changes.

Reply:
We should follow the semver guidelines here. Use major for breaking changes, minor for nonbreaking changes and patch for bug fixes only.

A simple mention that we do so and that this document is meant to elucidate what we consider breaking should suffice.

Reply:
Done.